lib/Doctrine/DBAL/RewindableGenerator.php

```diff
@@ -0,0 +1,23 @@
+<?php
+
+namespace Doctrine\DBAL;
+
+class RewindableGenerator implements \IteratorAggregate
+{
+    private $generator;
+
+    /**
+     * @param callable $generator
+     */
+    public function __construct(callable $generator)
+    {
+        $this->generator = $generator;
+    }
+
+    public function getIterator()
+    {
+        $g = $this->generator;
```

Comment:
This would simply call the generator callable at each `foreach () {}` (because each `foreach` will cause a `->getIterator()` call).

That's most likely what we do NOT want, as a query will be repeated.

The test case is as following:

```php
$result = someQuery();

foreach ($result as $value) {
}

foreach ($result as $value) {
}

self::assertCount(1, $executedQueries);
```